packages/x-data-grid-pro/src/hooks/features/columnResize/useGridColumnResize.tsx

```diff
@@ -378,6 +379,20 @@ export const useGridColumnResize = (
     // eslint-disable-next-line @typescript-eslint/no-use-before-define
     stopListening();
 
+    if (previousMouseClickEvent.current) {
+      const prevTimeStamp = previousMouseClickEvent.current.timeStamp;
+      const prevClientX = previousMouseClickEvent.current.clientX;
+      const prevClientY = previousMouseClickEvent.current.clientY;
+      if (
+        nativeEvent.timeStamp - prevTimeStamp < 300 &&
+        nativeEvent.clientX === prevClientX &&
+        prevClientY === nativeEvent.clientY
+      ) {
```

Comment:
Can you explain the logic behind this logic?

Reply:
So overall logic is. 

Mouse double click is combination of 2 mouseup and mousedown event. 

But according to our previous logic whenever mousedown event used to occur we add the event listeners to listen mouseup event and resize according. 


This logic had a collision with the mouse double click so just to distinguish between mouse double click and mouse up/ mouse down event I compared the time interval between mouse down and mouseup event and their coordinate if it is same this directly means user wants to do double click. 

Hope I am able to explain this

Reply:
@romgrk if you can kindly check